### PR TITLE
Laboratory work#2 Shevchenko Ksenya 17FPL1

### PR DESCRIPTION
**Pull request by KsenyaSH**

### CODE REVIEW

File: lab_2/main.py [hunk 1]
```diff
@@ -11,3 +11,98 @@
     with open('very_big_reference_text.txt', 'r') as f:
         REFERENCE_TEXT = f.read()
         freq_dict = calculate_frequences(REFERENCE_TEXT)
+def propose_candidates(word: str, max_depth_permutations: int=1) -> list:
```
Comment (mafatekhov) [inline lab_2/main.py]: @KsenyaSH Please define all functions outside of 'if main' condition

**Comment by mafatekhov** [comment]
@KsenyaSH 1 test is failing: lab_2.spell_check_word_test.SpellCheckWordsTest#test_spell_check_word_functions_called_for_incorrect_word

seems like keep_known function call is missed in spell_check_word function

**Comment by mafatekhov** [comment]
@KsenyaSH  spell_check_word is almost identical with @irenemfzlv - you even have the same error

**Comment by mafatekhov** [comment]
@KsenyaSH almost identical functions choose_best with @maria1299

**Comment by KsenyaSH** [comment]
@mafatekhov keep_known function is called in choose_best function, which is called in spell_check_word. If it's necessary for passing tests, it can be added in spell_check_word

**Pull request closed without merging**